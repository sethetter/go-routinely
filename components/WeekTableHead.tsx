import * as React from 'react'

const daysOfWeek = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT']

const WeekTableHead = () => {
  return (
    <thead className="thead-light">
      <tr>
        <th />
        {daysOfWeek.map(d => <th key={d}>{d}</th>)}
      </tr>
    </thead>
  )
}

export default WeekTableHead
