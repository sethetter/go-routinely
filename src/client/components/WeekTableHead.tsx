import * as React from 'react'
import moment from 'moment'

const daysOfWeek = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT']

export interface WeekTableHeadProps {
  startOfWeek: Date
}

const WeekTableHead = ({ startOfWeek }: WeekTableHeadProps) => {
  const sunday = moment(startOfWeek).startOf('week').format('M/D')
  const saturday = moment(startOfWeek).endOf('week').format('M/D')
  const weekRange = `${sunday} - ${saturday}`

  return (
    <thead className="thead-light">
      <tr>
        <th>{weekRange}</th>
        {daysOfWeek.map(d => <th key={d}>{d}</th>)}
      </tr>
    </thead>
  )
}

export default WeekTableHead
