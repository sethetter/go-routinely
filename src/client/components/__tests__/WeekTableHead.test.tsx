import * as React from 'react'
import * as moment from 'moment'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import WeekTableHead from '../WeekTableHead'

const startOfWeek = moment(new Date('03/25/2018')).startOf('week').toDate()

describe('WeekTableHead', () => {
  it('should match snapshot', () => {
    const output = shallow(<WeekTableHead startOfWeek={startOfWeek} />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })

  it('should render 8 th elements', () => {
    const output = shallow(<WeekTableHead startOfWeek={startOfWeek} />)
    expect(output.find('th').length).toBe(8)
  })

  it('should have each day of the week rendered', () => {
    const output = shallow(<WeekTableHead startOfWeek={startOfWeek} />)
    const expectedDays = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT']

    expectedDays.map(d => {
      const thForDay = output.findWhere(n => n.is('th') && n.text() === d)
      expect(thForDay).toHaveLength(1)
    })
  })
  
  it('shows the current week range', () => {
    const output = shallow(<WeekTableHead startOfWeek={startOfWeek} />)

    const sundayR = new RegExp(moment(startOfWeek).format('M/D'))
    const saturdayR = new RegExp(moment(startOfWeek).day('saturday').format('M/D'))

    const weekRangeNode = output.findWhere(n => {
      return n.is('th') && sundayR.test(n.text()) && saturdayR.test(n.text())
    })

    expect(weekRangeNode).toHaveLength(1)
  })
})
