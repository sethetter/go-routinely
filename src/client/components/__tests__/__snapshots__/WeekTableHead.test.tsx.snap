// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekTableHead should match snapshot 1`] = `
<thead
  className="thead-light"
>
  <tr>
    <th>
      3/25 - 3/31
    </th>
    <th
      key="SUN"
    >
      SUN
    </th>
    <th
      key="MON"
    >
      MON
    </th>
    <th
      key="TUE"
    >
      TUE
    </th>
    <th
      key="WED"
    >
      WED
    </th>
    <th
      key="THU"
    >
      THU
    </th>
    <th
      key="FRI"
    >
      FRI
    </th>
    <th
      key="SAT"
    >
      SAT
    </th>
  </tr>
</thead>
`;
